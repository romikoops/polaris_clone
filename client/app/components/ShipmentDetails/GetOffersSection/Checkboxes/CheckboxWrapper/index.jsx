import React from 'react'
import Checkbox from '../../../../Checkbox/Checkbox'

function CheckboxWrapper ({
  id, name, className, theme, checked, labelContent, onChange, show, style
}) {
  if (!show) return ''

  return (
    <div
      className={`${className} flex-100 layout-row layout-align-start-center`}
      style={style}
    >
      <div className="flex-10 layout-row layout-align-start-start">
        <Checkbox
          id={id}
          theme={theme}
          onChange={onChange}
          size="30px"
          name={name}
          checked={checked}
        />
      </div>
      <div className="flex">
        <label htmlFor={id} className="pointy">
          {labelContent}
        </label>
      </div>
    </div>
  )
}

export default CheckboxWrapper
