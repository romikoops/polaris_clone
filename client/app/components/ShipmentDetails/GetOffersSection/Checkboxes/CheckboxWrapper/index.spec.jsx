import * as React from 'react'
import { shallow } from 'enzyme'
import CheckboxWrapper from '.'
import {
  identity,
  theme
} from '../../../mocks'

const propsBase = {
  id: 1,
  theme,
  name: 'NAME',
  className: 'CLASS_NAME',
  checked: true,
  labelContent: 'LABEL_CONTENT',
  onChange: identity,
  show: true,
  style: {}
}

test('with empty props', () => {
  expect(shallow(<CheckboxWrapper />)).toMatchSnapshot()
})

test('happy path', () => {
  expect(shallow(<CheckboxWrapper {...propsBase} />)).toMatchSnapshot()
})
