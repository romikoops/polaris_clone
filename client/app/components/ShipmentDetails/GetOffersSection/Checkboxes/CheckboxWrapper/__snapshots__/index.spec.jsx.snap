// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`happy path 1`] = `
<div
  className="CLASS_NAME flex-100 layout-row layout-align-start-center"
  style={Object {}}
>
  <div
    className="flex-10 layout-row layout-align-start-start"
  >
    <Checkbox
      checked={true}
      className=""
      disabled={false}
      id={1}
      name="NAME"
      onChange={null}
      onClick={null}
      size="30px"
      theme={
        Object {
          "background": "https://assets.itsmycargo.com/assets/images/cropped_banner_2.jpg",
          "colors": Object {
            "brightPrimary": "#2491FD",
            "brightSecondary": "#25ED36",
            "primary": "#0D5BA9",
            "secondary": "#23802A",
          },
          "logoLarge": "https://assets.itsmycargo.com/assets/logos/logo_box.png",
          "logoSmall": "https://assets.itsmycargo.com/assets/logos/logo_box.png",
          "logoWhite": "https://assets.itsmycargo.com/assets/logos/Logo_transparent_white.png",
          "logoWide": "https://assets.itsmycargo.com/assets/logos/Logo_transparent_blue.png",
        }
      }
    />
  </div>
  <div
    className="flex"
  >
    <label
      className="pointy"
      htmlFor={1}
    >
      LABEL_CONTENT
    </label>
  </div>
</div>
`;

exports[`with empty props 1`] = `""`;
